### PR TITLE
Rename Spring Profile dev to debug

### PR DESCRIPTION
**Pull request by kilbphilippSAP**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn install` runs for the whole project and, if you touched any code in the respective service, submission and distribution service can be run with `spring-boot:run`

## Description

* Rename spring-profile 'dev' to 'debug'
* Adjusted documentation accordingly

Linked to: [CWA-Admin PR](https://github.com/corona-warn-app/cwa-admin/pull/110)

### CODE REVIEW

**Review by ioangut — COMMENTED** [review]
I would say that this PR should be for the moment in draft status as there are other documentation PRs that includes  or add the spring profile(Docs/federation #861) and those should be updated too.

**Review by emmetsap — DISMISSED** [review]

**Comment by ioangut** [comment]
As the documentation PR were merged, these PR can be updated

**Review by mibrasap — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -150,10 +150,10 @@ If you want to start the submission service, for example, you start it as follow
 
 #### Debugging
 
-To enable the `DEBUG` log level, you can run the application using the Spring `dev` profile.
+To enable the `DEBUG` log level, you can run the application using the Spring `debug` profile.
```
Comment (hilmarf) [inline README.md:153]: @fredrb / @kilbphilippSAP - I thought, we wanted to name it 'log'!?! or loglevel or debug-log

**Review by hilmarf — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (kilbphilippSAP) [reply README.md:153]: @hilmarf / @fredrb - The task did not specify a name ("Rename dev Spring profile to something like `logging`, `debug` or another more meaningful name for raising the log level."), so I went with `debug`. I can change it to something else, if so desired.

**Pull request merged**